CommunicationLayer/MessagingFramework/MessageDecodingHelpers.cpp

```diff
@@ -53,7 +70,14 @@ unsigned char MessageDecodingHelpers::getUnsignedChar(const QByteArray& data, in
 
 unsigned int MessageDecodingHelpers::getUnsignedInt(const QByteArray& data, int startIndex)
 {
-    return data.mid(startIndex, numberOfBytesInData(Type::UNSIGNED_INT)).toUInt();
+    unsigned int value;
+
+    QDataStream dataStream(data.mid(startIndex, numberOfBytesInData(Type::UNSIGNED_INT)));
```

Comment:
The only difference between the 3 functions really... are the type of return, and then `Type::UNSIGNED_INT` `Type::UNSIGNED_SHORT` etc

There's a couple ways you could go about getting rid of the copy pasta code :P 

Using one Template to completely replace all 3 functions (hardest part would be matching the `T` value to find out what `Type::<type>` to use)

Or probably easier: make use of a helper function that uses a Template

```
//Helper function
template <class T>
void readQDataStream(QDataStream dataStream, T &var) {
   dataStream.setByteOrder(QDataStream::LittleEndian);
   dataStream >> var;
}
// Your normal function
unsigned int MessageDecodingHelpers::getUnsignedInt(const QByteArray& data, int startIndex) {
    unsigned int value;
    QDataStream dataStream(data.mid(startIndex, numberOfBytesInData(Type::UNSIGNED_INT)));
    readQDataStream<unsigned int>(dataStream, value);
    return value;
}

float MessageDecodingHelpers::getFloat(const QByteArray& data, int startIndex) {
    float value;
    QDataStream dataStream(data.mid(startIndex, numberOfBytesInData(Type::FLOAT)));
    readQDataStream<float>(dataStream, value);
    return value;
}
```

Not sure if worth it :P, and not sure if it works 100%. But could be a good opportunity to make use of your c++ knowledge.